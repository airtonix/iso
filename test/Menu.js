const React = require('react')
const { create: render } = require('react-test-renderer')

jest.mock('electron', () => ({
  app: {},
  remote: {
    app: {
      getName: () => 'Test',
      getPath: () => ''
    },
    dialog: {
      showSaveDialog: jest.fn(),
      showOpenDialog: jest.fn()
    }
  }
}))

global.Notification = {}

const dialogs = require('../src/dialogs')
jest.mock('../src/dialogs', () => ({
  newFile: jest.fn(),
  openFile: jest.fn()
}))

jest.mock('../src//updaters', () => ({
  openFile: () => {}
}))

const { default: Menu, config } = require('../src/Menu')

describe('Menu', () => {
  test('renders null', () => {
    const json = render(
      React.createElement(Menu, {})
    ).toJSON()
    expect(json).toBe(null)
  })

  test('config returns an array', () => {
    const menu = config({})
    expect(Array.isArray(menu)).toBe(true)
  })

  test('config snapshot', () => {
    const menu = config({})
    expect(menu).toMatchSnapshot()
  })

  describe('click handlers', () => {
    let menu, app, file, view

    beforeAll(() => {
      menu = config({})
      app = menu[0]
      file = menu[1]
      view = menu[3]
    })

    test('File > New opens dialog', () => {
      const [ newFile ] = file.submenu
      newFile.click()
      expect(dialogs.newFile).toHaveBeenCalled()
    })

    test('File > Open opens dialog', () => {
      const [ , openFile ] = file.submenu
      openFile.click()
      expect(dialogs.openFile).toHaveBeenCalled()
    })

    test.skip('File > Open Recent opens dialog')
    test.skip('File > Open Recent > Clear List updates recents')
    test.skip('File > Open Recent > Clear List updates recents')

    test.skip('View > Hide Panel updates editorOpen')
    test.skip('View > Dark Mode updates store.uiColor')
    test.skip('View > Light Mode updates store.uiColor')
    test.skip('View > VIM Mode updates store.vimMode')
    test.skip('View > Enable Styled Components updates store.styledComponents')

    test.skip('Help > Getting Started updates filename')
  })
})
