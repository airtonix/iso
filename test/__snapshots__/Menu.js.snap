// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu config snapshot 1`] = `
Array [
  Object {
    "label": "Test",
    "submenu": Array [
      Object {
        "role": "about",
      },
      Object {
        "click": [Function],
        "label": "Check for Updates...",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "services",
        "submenu": Array [],
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "hide",
      },
      Object {
        "role": "hideothers",
      },
      Object {
        "role": "unhide",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "quit",
      },
    ],
  },
  Object {
    "label": "File",
    "submenu": Array [
      Object {
        "accelerator": "Cmd+N",
        "click": [Function],
        "label": "New...",
      },
      Object {
        "accelerator": "Cmd+O",
        "click": [Function],
        "label": "Open...",
      },
      Object {
        "label": "Open Recent",
        "submenu": Array [
          Object {
            "type": "separator",
          },
          Object {
            "click": [Function],
            "label": "Clear List",
          },
        ],
      },
      Object {
        "type": "separator",
      },
      Object {
        "click": [Function],
        "enabled": false,
        "label": "Open in Editor",
      },
      Object {
        "click": [Function],
        "label": "Export HTML",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "close",
      },
    ],
  },
  Object {
    "role": "editMenu",
  },
  Object {
    "label": "View",
    "submenu": Array [
      Object {
        "accelerator": "Cmd+E",
        "click": [Function],
        "enabled": false,
        "label": "View Folder",
        "visible": false,
      },
      Object {
        "accelerator": "Cmd+3",
        "click": [Function],
        "label": "Hide Panel",
      },
      Object {
        "type": "separator",
      },
      Object {
        "checked": false,
        "click": [Function],
        "label": "Dark Mode",
        "type": "checkbox",
      },
      Object {
        "checked": false,
        "click": [Function],
        "label": "Light Mode",
        "type": "checkbox",
      },
      Object {
        "type": "separator",
      },
      Object {
        "checked": undefined,
        "click": [Function],
        "label": "VIM Mode",
        "type": "checkbox",
      },
      Object {
        "checked": undefined,
        "click": [Function],
        "label": "Enable Styled Components",
        "type": "checkbox",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "resetzoom",
      },
      Object {
        "role": "zoomin",
      },
      Object {
        "role": "zoomout",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "togglefullscreen",
      },
    ],
  },
  Object {
    "role": "windowMenu",
  },
  Object {
    "role": "help",
    "submenu": Array [
      Object {
        "click": [Function],
        "label": "Getting Started",
      },
      Object {
        "click": [Function],
        "label": "Test Help",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "toggledevtools",
      },
      Object {
        "accelerator": "Cmd+Shift+R",
        "role": "reload",
      },
    ],
  },
]
`;
